src/app/backend/replicasetcommon.go

```diff
@@ -100,3 +114,78 @@ func getReplicaSetPodInfo(replicaSet *api.ReplicationController, pods []api.Pod)
 
 	return result
 }
+
+// Returns Pod Events of type warning from given list of pods
+// TODO(floreks) : use some set to get only unique error messages
+func GetPodsEventErrors(client *client.Client, pods []api.Pod) (result []PodEvent, err error) {
+	log.Printf("Getting list of pods error events")
+
+	for _, pod := range pods {
+		if !isRunningOrSucceeded(pod) {
+			events, err := GetPodEvents(client, pod)
+
+			if err != nil {
+				return nil, err
+			}
+
+			result = getPodsEventErrors(events)
+		}
+	}
+
+	return removeDuplicates(result), nil
+}
+
+// Filters kubernetes API event objects based on event type. Empty string will return all events
+func FilterEventsByType(events []api.Event, eventType string) []api.Event {
+	if len(eventType) == 0 {
+		return events
+	}
+
+	result := make([]api.Event, 0)
+	for _, event := range events {
+		if event.Type == eventType {
+			result = append(result, event)
+		}
+	}
+
+	return result
+}
+
+// Returns list of Pod Event model objects based on kubernetes API event objects
```

Comment:
You are right. It's actually based on kubernetes API EventList object. I'll change.